src/tests/suite/daos_container.c

```diff
@@ -771,6 +771,226 @@ co_set_prop(void **state)
 	test_teardown((void **)&arg);
 }
 
+static void
+co_create_access_denied(void **state)
+{
+	test_arg_t	*arg0 = *state;
+	test_arg_t	*arg = NULL;
+	daos_prop_t	*prop;
+	int		 rc;
+
+	rc = test_setup((void **)&arg, SETUP_EQ, arg0->multi_rank,
+			DEFAULT_POOL_SIZE, NULL);
+	assert_int_equal(rc, 0);
+
+	print_message("Try to create container on pool with no create perms\n");
+
+	/* on the pool, write is an alias for create+del cont */
+	prop = get_daos_prop_with_owner_acl_perms(DAOS_ACL_PERM_POOL_ALL &
+						  ~DAOS_ACL_PERM_CREATE_CONT &
+						  ~DAOS_ACL_PERM_WRITE,
+						  DAOS_PROP_PO_ACL);
+
+	while (!rc && arg->setup_state != SETUP_POOL_CONNECT)
+		rc = test_setup_next_step((void **)&arg, NULL, prop, NULL);
+	assert_int_equal(rc, 0);
+
+	MPI_Barrier(MPI_COMM_WORLD);
+
+	uuid_generate(arg->co_uuid);
+	rc = daos_cont_create(arg->pool.poh, arg->co_uuid, NULL, NULL);
+	assert_int_equal(rc, -DER_NO_PERM);
+
+	MPI_Barrier(MPI_COMM_WORLD);
+
+	daos_prop_free(prop);
+	test_teardown((void **)&arg);
+}
+
+static void
+co_destroy_access_denied(void **state)
+{
+	test_arg_t	*arg0 = *state;
+	test_arg_t	*arg = NULL;
+	daos_prop_t	*pool_prop;
+	daos_prop_t	*cont_prop;
+	int		 rc;
+	struct daos_acl	*cont_acl;
+	struct daos_ace	*update_ace;
+	daos_handle_t	coh;
+
+	rc = test_setup((void **)&arg, SETUP_EQ, arg0->multi_rank,
+			DEFAULT_POOL_SIZE, NULL);
+	assert_int_equal(rc, 0);
+
+	print_message("Try to delete container where pool and cont deny\n");
+
+	/*
+	 * Pool doesn't give the owner delete cont privs. For the pool, write
+	 * is an alias for create+del container.
+	 */
+	pool_prop = get_daos_prop_with_owner_acl_perms(DAOS_ACL_PERM_POOL_ALL &
+						       ~DAOS_ACL_PERM_DEL_CONT &
+						       ~DAOS_ACL_PERM_WRITE,
+						       DAOS_PROP_PO_ACL);
+
+	/* container doesn't give delete privs to the owner */
+	cont_prop = get_daos_prop_with_owner_acl_perms(DAOS_ACL_PERM_CONT_ALL &
+						       ~DAOS_ACL_PERM_DEL_CONT,
+						       DAOS_PROP_CO_ACL);
+
+	while (!rc && arg->setup_state != SETUP_CONT_CREATE)
+		rc = test_setup_next_step((void **)&arg, NULL, pool_prop,
+					  cont_prop);
+	assert_int_equal(rc, 0);
+
+	MPI_Barrier(MPI_COMM_WORLD);
+
+	rc = daos_cont_destroy(arg->pool.poh, arg->co_uuid, 1, NULL);
```

Comment:
should the container destroy call be done only on rank 0 if only rank 0 was the one to create the container in the test_setup_next_step execution above?

Reply:
Good catch on this. A lot of this new test code was monkey-see-monkey-do based on existing tests, and I wasn't aware of some of these subtleties. I'm re-reviewing these tests to look more closely at this.